RHIDP-3878 Authenticating with RHSSO

themr0c opened the pull request:
**Version(s):** 1.3

**Issue:** https://issues.redhat.com/browse/RHIDP-3878


**Link to docs preview:** https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-480/authentication/#assembly-authenticating-with-rhsso

**Reviews:**
- [ ] SME: @ mention assignee
- [x] QE: @albarbaro
- [ ] Docs review: @ mention assignee
<!--- SME approval is required to merge a PR unless the changes are made by a subject matter expert. --->
<!--- QE approval is required to merge a PR unless there are no technical changes to the content. --->
<!--- Docs team approval is required for ALL PRs. --->

**Additional information:**
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, request reviews from all required stakeholders via Slack GitHub, or Jira. --->

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,71 @@
+You might use this authentication provider to authenticate with other third-party services that support the OIDC protocol.
+
+.Prerequisites
+* You added a custom {product-short} application configuration, and have sufficient permissions to modify it.
+See link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html/administration_guide_for_red_hat_developer_hub/assembly-add-custom-app-file-openshift_admin-rhdh[Adding a custom application configuration file to {product-short}].
```
```suggestion
* You link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html/administration_guide_for_red_hat_developer_hub/assembly-add-custom-app-file-openshift_admin-rhdh[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+* You have sufficient permissions in RHSSO to create and manage a realm.
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO:
```
```suggestion
. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user], and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short} application]:
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Create a new realm, such as __<{product-very-short}>__.
```
```suggestion
.. Create a new realm, with a distinctive **Name** such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Create a new realm, with a distinctive **Name** such as __<{product-very-short}>__.
```
```suggestion
.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[Create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[Create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
```
```suggestion
.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
+.. In this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with a **Username** such as __<rhdhuser>__, and toggle **Email verified** to `On`.
+.. To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
+.. Create a new Client ID, such as __<{product-very-short}>__.
```
```suggestion
.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with a distinctive **CLient ID** such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,70 @@
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
+.. In this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with a **Username** such as __<rhdhuser>__, and toggle **Email verified** to `On`.
```
```suggestion
.. In this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
... **Email verified**: Toggle to `On`.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
+... **Email verified**: Toggle to `On`.
+.. To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
+.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with a distinctive **CLient ID** such as __<{product-very-short}>__.
```
```suggestion
.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with:
... **CLient ID**: A distinctive client ID, such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,73 @@
+.. To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
+.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with:
+... **CLient ID**: A distinctive client ID, such as __<{product-very-short}>__.
+*** **Client authentication**: Toggle to On.
+*** **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
```
```suggestion
//... **Client authentication**: Toggle to On.
... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,73 @@
+... **CLient ID**: A distinctive client ID, such as __<{product-very-short}>__.
+//... **Client authentication**: Toggle to On.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+.. Navigate to the **Credentials** tab and copy the **Client secret**.
+.. Save the **Client ID** and the **Client Secret** values for the next step.
```
```suggestion
... Navigate to the **Credentials** tab and copy the **Client secret**.
.. Save the **Client ID** and the **Client Secret** values for the next step.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,73 @@
+... **Email verified**: Toggle to `On`.
+.. To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
+.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with:
+... **CLient ID**: A distinctive client ID, such as __<{product-very-short}>__.
```
```suggestion
... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,73 @@
+//... **Client authentication**: Toggle to On.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+.. Save the **Client ID** and the **Client Secret** values for the next step.
```
```suggestion
.. Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,73 @@
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
+.. In this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
+... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
+... **Email verified**: Toggle to `On`.
+.. To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
```
```suggestion
.. In this realm, use an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
... **Email verified**: Toggle to `On`.
... To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
... Save the **Username** and **Password** for the verification steps.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,74 @@
+.Verification
+. Go to the {product-short} login page.
+. Your {product-short} sign-in page displays *Sign in using OIDC* and the Guest user sign-in is disabled.
+. Log in with OIDC.
```
```suggestion
. Log in with OIDC by using the saved **Username** and **Password** values.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,74 @@
++
+`AUTH_OIDC_CLIENT_ID`:: Enter the Client ID that you generated in Keycloak, such as `RHDH`.
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the Client Secret that you generated in Keycloak.
+`AUTH_OIDC_METADATA_URL`:: Enter your RHSSO realm base URL, such as: __<your_rhsso_URL>__/auth/realms/__<{product-very-short}>__
```
```suggestion
`AUTH_OIDC_METADATA_URL`:: Enter your RHSSO realm base URL, such as: __<your_rhsso_URL>__/auth/realms/__<{product-very-short}>__.

```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,75 @@
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+.. Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
+
+. Add your RHSSO credentials in your Developer Hub secrets.
+.. Edit your Developer Hub secrets, such as secrets-rhdh, and add the following key/value pairs:
```
```suggestion
. To add your RHSSO credentials in your Developer Hub secrets, edit your Developer Hub secrets, such as `secrets-rhdh`, and add the following key/value pairs:
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,74 @@
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the Client Secret that you generated in Keycloak.
+`AUTH_OIDC_METADATA_URL`:: Enter your RHSSO realm base URL, such as: __<your_rhsso_URL>__/auth/realms/__<{product-very-short}>__.
+
+. Set up the GitHub authentication provider in your {product-short} custom configuration.
+Edit your custom Developer Hub config map, such as `app-config-rhdh`.
+In the `app-config-rhdh.yaml` content, add the `oidc` provider configuration,
+and enable the `oidc` provider for sign-in:
```
```suggestion
. To set up the RHSSO authentication provider in your {product-short} custom configuration, edit your custom Developer Hub config map, such as `app-config-rhdh`, and add following lines to the `app-config-rhdh.yaml` content:
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,71 @@
++
+.`app-config-rhdh.yaml` fragment
+[source,yaml]
+----
+auth:
+  environment: production # <1>
+  providers:
+    oidc:
+      production:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
+        prompt: auto
+        ## Uncomment for additional configuration options
+        # callbackUrl: ${AUTH_OIDC_CALLBACK_URL}
+        # tokenEndpointAuthMethod: ${AUTH_OIDC_TOKEN_ENDPOINT_METHOD}
+        # tokenSignedResponseAlg: ${AUTH_OIDC_SIGNED_RESPONSE_ALG}
+        # scope: ${AUTH_OIDC_SCOPE}
+        ## Declarative resolvers to override the default resolver: `emailLocalPartMatchingUserEntityName`
+        ## The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed. Uncomment the resolvers that you want to use.
+        # signIn:
+        #   resolvers:
+        #     - resolver: preferredUsernameMatchingUserEntityName
+        #     - resolver: emailMatchingUserEntityProfileEmail
+        #     - resolver: emailLocalPartMatchingUserEntityName
+signInPage: oidc # <2>
+dangerouslyAllowSignInWithoutUserInCatalog: true # <3>
+----
++
+<1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.
+<2> To enable the OIDC provider as default sign-in provider.
+<3> To enable authentication without requiring to provision the user catalog.
```
```suggestion
----
auth:
  environment: production # <1>
  providers:
    oidc:
      production:
        metadataUrl: ${AUTH_OIDC_METADATA_URL} # <2>
        clientId: ${AUTH_OIDC_CLIENT_ID} # <2>
        clientSecret: ${AUTH_OIDC_CLIENT_SECRET} # <2>
        prompt: auto
        ## Uncomment for additional configuration options
        # callbackUrl: ${AUTH_OIDC_CALLBACK_URL}
        # tokenEndpointAuthMethod: ${AUTH_OIDC_TOKEN_ENDPOINT_METHOD}
        # tokenSignedResponseAlg: ${AUTH_OIDC_SIGNED_RESPONSE_ALG}
        # scope: ${AUTH_OIDC_SCOPE}
        ## Declarative resolvers to override the default resolver: `emailLocalPartMatchingUserEntityName`
        ## The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed. Uncomment the resolvers that you want to use.
        # signIn:
        #   resolvers:
        #     - resolver: preferredUsernameMatchingUserEntityName
        #     - resolver: emailMatchingUserEntityProfileEmail
        #     - resolver: emailLocalPartMatchingUserEntityName
signInPage: oidc # <3>
dangerouslyAllowSignInWithoutUserInCatalog: true # <4>
----
+
<1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.
<2> To configure the OIDC provider with your secrets.
<3> To enable the OIDC provider as default sign-in provider.
<4> To enable authentication without requiring to provision the user catalog.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+        metadataUrl: ${AUTH_OIDC_METADATA_URL} # <2>
+        clientId: ${AUTH_OIDC_CLIENT_ID} # <2>
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET} # <2>
+        prompt: auto
```
```suggestion
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,71 @@
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
+.. In this realm, use an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
+... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
+... **Email verified**: Toggle to `On`.
+... To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
+... Save the **Username** and **Password** for the verification steps.
+.. link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[Create a Client ID], with:
+... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
+//... **Client authentication**: Toggle to On.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+.. Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
```
```suggestion
.. To register your {product-short} in RHSSO, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
//... **Client authentication**: Toggle to On.
... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
... Navigate to the **Credentials** tab and copy the **Client secret**.
... Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.

.. To prepare for the verification steps, in this realm, use an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
... **Email verified**: Toggle to `On`.
... To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
... Save the **Username** and **Password** for the verification steps.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
+.. To register your {product-short} in RHSSO, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
```
```suggestion
.. To register your {product-short} in RHSSO, in this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
+
+.. To prepare for the verification steps, in this realm, use an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
```
```suggestion
.. To prepare for the verification steps, in the same realm, get the credential information for an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
```

josephca commented:
@albarbaro can you please provide QE review? thanks!

kim-tsao commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+
+. To add your RHSSO credentials in your Developer Hub secrets, edit your Developer Hub secrets, such as `secrets-rhdh`, and add the following key/value pairs:
++
+`AUTH_OIDC_CLIENT_ID`:: Enter the Client ID that you generated in Keycloak, such as `RHDH`.
```
Should we replace all occurrences of Keycloak with RHSSO?

themr0c replied:
Absolutely!

albarbaro commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,9 @@
+
+To authenticate users with Red Hat SSO (RHSSO), enable the OpenID Connect (OIDC) authentication provider in RHDH.
+
+You might use this authentication provider to authenticate with other third-party services that support the OIDC protocol.
```
I agree with Kim, this is not tested.

themr0c replied:
Removed

themr0c commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,9 @@
+= Authenticating with Red Hat Single Sign-On (RHSSO)
+
+To authenticate users with Red Hat SSO (RHSSO), enable the OpenID Connect (OIDC) authentication provider in RHDH.
+
+You might use this authentication provider to authenticate with other third-party services that support the OIDC protocol.
```
```suggestion
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+
+To authenticate users with Red Hat SSO (RHSSO), enable the OpenID Connect (OIDC) authentication provider in {product}.
+
+You might use this authentication provider to authenticate with other third-party services that support the OIDC protocol.
```
```suggestion
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,71 @@
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
+
+.. To prepare for the verification steps, in the same realm, get the credential information for an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
+... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
+... **Email verified**: Toggle to `On`.
+... To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
```
Indeed, we should not repeat the RHSSO docs:

```suggestion
.. To prepare for the verification steps, in the same realm, get the credential information for an existing user or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user and save the user credential information].
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,68 @@
+
+. To add your RHSSO credentials in your Developer Hub secrets, edit your Developer Hub secrets, such as `secrets-rhdh`, and add the following key/value pairs:
++
+`AUTH_OIDC_CLIENT_ID`:: Enter the Client ID that you generated in Keycloak, such as `RHDH`.
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the Client Secret that you generated in Keycloak.
+`AUTH_OIDC_METADATA_URL`:: Enter your RHSSO realm base URL, such as: __<your_rhsso_URL>__/auth/realms/__<{product-very-short}>__.
```
```suggestion
`AUTH_OIDC_CLIENT_ID`:: Enter the saved **Client ID**.
`AUTH_OIDC_CLIENT_SECRET`:: Enter the saved **Client Secret**.
`AUTH_OIDC_METADATA_URL`:: Enter your **RHSSO realm base URL**, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,68 @@
+//... **Client authentication**: Toggle to On.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the **RHSSO URL**, **Realm name**, **Client ID**, and **Client Secret** values for the next step.
```
```suggestion
... Save the values for the next steps:
* **RHSSO realm base URL**
* **Client ID**
* **Client Secret**
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,68 @@
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<{product-very-short}>__.
```
```suggestion
.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<my_realm>__.
.. Save your **RHSSO realm base URL** for the next step, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+.. To prepare for the verification steps, in the same realm, get the credential information for an existing user, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user], with:
+... **Username**: Set a distinctive user name, such as __<rhdhuser>__.
+... **Email verified**: Toggle to `On`.
+... To set a password for this user, navigate to the **Credentials** tab, click **Set password**, enter the **Password**, and toggle **Temporary** to `Off`.
```
Understood

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+* You have sufficient permissions in RHSSO to create and manage a realm.
+
+.Procedure
+. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short ...
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<my_realm>__.
+.. Save your **RHSSO realm base URL** for the next step, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.
+.. To register your {product-short} in RHSSO, in this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
+... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
+//... **Client authentication**: Toggle to On.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the values for the next step:
+* **RHSSO realm base URL**
+* **Client ID**
+* **Client Secret**
+
+.. To prepare for the verification steps, in the same realm, get the credential information for an existing user or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user and save the user credential information].
+... Save the **Username** and **Password** for the verification steps.
```
```suggestion
. To allow Developer Hub to authenticate with RHSSO, complete the steps in RHSSO, to link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#realms-apps_[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[register the {product-short} application]:

.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<my_realm>__.
Save the value for the next step:
* **RHSSO realm base URL**, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.

.. To register your {product-short} in RHSSO, in this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
... Navigate to the **Credentials** tab and copy the **Client secret**.
... Save the values for the next step:
* **Client ID**
* **Client Secret**

.. To prepare for the verification steps, in the same realm, get the credential information for an existing user or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user]. Save the user credential information for the verification steps.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+
+.. Use an existing realm, or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-realm_[create a realm], with a distinctive **Name** such as __<my_realm>__.
+Save the value for the next step:
+* **RHSSO realm base URL**, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.
```
```suggestion
* **RHSSO realm base URL**, such as: __<your_rhsso_URL>__/auth/realms/__<your_realm>__.
```

themr0c commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
++
+`AUTH_OIDC_CLIENT_ID`:: Enter the saved **Client ID**.
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the saved **Client Secret**.
+`AUTH_OIDC_METADATA_URL`:: Enter your **RHSSO realm base URL**, such as: __<your_rhsso_URL>__/auth/realms/__<my_realm>__.
```
```suggestion
`AUTH_OIDC_METADATA_URL`:: Enter the saved **RHSSO realm base URL**.
```

kim-tsao commented on modules/auth/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,72 @@
+<1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.
+<2> To configure the OIDC provider with your secrets.
+<3> To enable the OIDC provider as default sign-in provider.
+<4> To enable authentication without requiring to provision the {product-short} software catalog.
```
```suggestion
<4> To enable authentication without requiring to provision users in the {product-short} software catalog.
```
This example goes though a non-prod set up.  Is there a follow up issue to show how to ingest users for a prod set up?  This would require refactoring the [instructions](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/configuring_plugins_in_red_hat_developer_hub/rhdh-installing-dynamic-plugins#rhdh-keycloak_plugin-rhdh) to set up keycloak plugin

themr0c commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,7 @@
+To authenticate users with Red Hat SSO (RHSSO), enable the OpenID Connect (OIDC) authentication provider in RHDH.
+
+include::modules/auth/proc-enabling-authentication-with-rhsso.adoc[leveloffset=+1]
+
```
Missing procedure: Provisioning users from RHSSO to the software catalog

themr0c commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,152 @@
+.Successful synchronization example:
+[source,json]
+----
+FIXME: add console log example
```
@albarbaro Can you provide a console log example?

themr0c replied:
```suggestion
{"class":"KeycloakOrgEntityProvider","level":"info","message":"Read 3 Keycloak users and 2 Keycloak groups in 1.5 seconds. Committing...","plugin":"catalog","service":"backstage","taskId":"KeycloakOrgEntityProvider:default:refresh","taskInstanceId":"bf0467ff-8ac4-4702-911c-380270e44dea","timestamp":"2024-09-25 13:58:04"}
{"class":"KeycloakOrgEntityProvider","level":"info","message":"Committed 3 Keycloak users and 2 Keycloak groups in 0.0 seconds.","plugin":"catalog","service":"backstage","taskId":"KeycloakOrgEntityProvider:default:refresh","taskInstanceId":"bf0467ff-8ac4-4702-911c-380270e44dea","timestamp":"2024-09-25 13:58:04"}
```

themr0c commented:
Resetted the approval count, as I introduced a new procedure.

themr0c commented:
@hmanwani-rh @PatAKnight @josephca @gustavolira @kim-tsao @albarbaro Can you review again, please? 
Since you reviewed:
* I introduced a new procedure: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-480/authentication/#provisioning-users-from-rhsso-to-the-software-catalog
* I added a warning on the `dangerouslyAllowSignInWithoutUserInCatalog: true` optional field in https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-480/authentication/#enabling-authentication-with-rhsso

themr0c commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,8 @@
+[id="assembly-authenticating-with-rhsso"]
+= Authenticating with Red Hat Single Sign-On (RHSSO)
+
+To authenticate users with Red Hat SSO (RHSSO), enable the OpenID Connect (OIDC) authentication provider in RHDH.
```
```suggestion
To authenticate users with Red Hat SSO (RHSSO): 
. xref:enabling-authentication-with-rhsso[Enable the OpenID Connect (OIDC) authentication provider in RHDH].
. xref:provisioning-users-from-rhsso-to-the-software-catalog[Provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog].
```

themr0c commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,10 @@
+[id="assembly-authenticating-with-rhsso"]
+= Authenticating with Red Hat Single Sign-On (RHSSO)
+
+To authenticate users with Red Hat SSO (RHSSO): 
```
```suggestion
To authenticate users with Red Hat SSO (RHSSO):

```

themr0c commented on assemblies/assembly-authenticating-with-rhsso.adoc:
```diff
@@ -0,0 +1,11 @@
+To authenticate users with Red Hat SSO (RHSSO):
+
+. xref:enabling-authentication-with-rhsso[Enable the OpenID Connect (OIDC) authentication provider in RHDH].
+. xref:provisioning-users-from-rhsso-to-the-software-catalog[Provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog].
```
```suggestion
. xref:provisioning-users-from-rhsso-to-the-software-catalog[Provision users from Red Hat Single-Sign On (RHSSO) to the software catalog].
```

kim-tsao reviewed (approved):
small nit but looks good otherwise

albarbaro reviewed (approved):
looks good to me.

themr0c commented:
/cherry-pick release-1.3

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the values for the next step:
```
```suggestion
... Save the values of **Client ID** and **Client Secret** for the next step.
```

themr0c replied:
This is less readable.

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the values for the next step:
+* **Client ID**
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the values for the next step:
+* **Client ID**
+* **Client Secret**
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+
+.. To prepare for the verification steps, in the same realm, get the credential information for an existing user or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user]. Save the user credential information for the verification steps.
+
+. To add your RHSSO credentials in your Developer Hub secrets, edit your Developer Hub secrets, such as `secrets-rhdh`, and add the following key/value pairs:
```
```suggestion
. To add your RHSSO credentials to your {product-short} secrets, edit your {product-short} secrets, such as `secrets-rhdh`, and add the following key/value pairs:
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the saved **Client Secret**.
+`AUTH_OIDC_METADATA_URL`:: Enter the saved **RHSSO realm base URL**.
+
+. To set up the RHSSO authentication provider in your {product-short} custom configuration, edit your custom Developer Hub config map, such as `app-config-rhdh`, and add following lines to the `app-config-rhdh.yaml` content:
```
```suggestion
. To set up the RHSSO authentication provider in your {product-short} custom configuration, edit your custom {product-short} ConfigMap such as `app-config-rhdh`, and add the following lines to the `app-config-rhdh.yaml` content:
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+. To set up the RHSSO authentication provider in your {product-short} custom configuration, edit your custom Developer Hub config map, such as `app-config-rhdh`, and add following lines to the `app-config-rhdh.yaml` content:
++
+--
+.`app-config-rhdh.yaml` fragment with mandatory fields to enable authentication with RHSSO
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with mandatory fields to enable authentication with RHSSO
```

themr0c replied:
For minimalism, fragment is enough, example would be a répétition.

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+[source,yaml]
+----
+auth:
+  environment: production
```
```suggestion
  environment: production <1>
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+----
+
+`environment: production`::
+Mark the environment as `production` to hide the Guest login in the {product-short} home page.
```
```suggestion
<1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+signInPage: oidc
+----
+
+`environment: production`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+  providers:
+    oidc:
+      production:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
```
```suggestion
        metadataUrl: ${AUTH_OIDC_METADATA_URL} <2>
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+    oidc:
+      production:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
```
```suggestion
        clientId: ${AUTH_OIDC_CLIENT_ID} <2>
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+      production:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
```
```suggestion
        clientSecret: ${AUTH_OIDC_CLIENT_SECRET} <2>
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+`environment: production`::
+Mark the environment as `production` to hide the Guest login in the {product-short} home page.
+
+`metadataUrl`, `clientId`, `clientSecret`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+Mark the environment as `production` to hide the Guest login in the {product-short} home page.
+
+`metadataUrl`, `clientId`, `clientSecret`::
+To configure the OIDC provider with your secrets.
```
```suggestion
<2> To configure the OIDC provider with your secrets.
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
+signInPage: oidc
```
```suggestion
signInPage: oidc <3>
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+`metadataUrl`, `clientId`, `clientSecret`::
+To configure the OIDC provider with your secrets.
+
+`sigInPage: oidc`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+To configure the OIDC provider with your secrets.
+
+`sigInPage: oidc`::
+To enable the OIDC provider as default sign-in provider.
```
```suggestion
<3> To enable the OIDC provider as the default sign-in provider.
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+To enable the OIDC provider as default sign-in provider.
+
+
+Optional: Consider adding following optional fields:
```
```suggestion
Optional: Consider adding the following optional fields:
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
++
+WARNING: Use this option to explore {product-short} features, but do not use it in production.
++
+.`app-config-rhdh.yaml` fragment with optional field to allow authenticating users absent from the software catalog
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with an optional field to allow authenticating users absent from the software catalog
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+--
+RHSSO callback URL.
+
+.`app-config-rhdh.yaml` fragment with optional `callbackURL` field
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with optional `callbackURL` field
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+--
+Token endpoint authentication method.
+
+.`app-config-rhdh.yaml` fragment with optional `tokenEndpointAuthMethod` field
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with optional `tokenEndpointAuthMethod` field
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+--
+Token signed response algorithm.
+
+.`app-config-rhdh.yaml` fragment with optional `tokenSignedResponseAlg` field
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with optional `tokenSignedResponseAlg` field
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+--
+RHSSO scope.
+
+.`app-config-rhdh.yaml` fragment with optional `scope` field
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with optional `scope` field
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+
+`signIn.resolvers`::
+--
+Declarative resolvers to override the default resolver: `emailLocalPartMatchingUserEntityName`.
```
```suggestion
Declarative resolvers can be used to override the default resolver: emailLocalPartMatchingUserEntityName. The authentication provider attempts each sign-in resolver in sequence until one succeeds and fails if none do.
```

hmanwani-rh commented on modules/authentication/proc-enabling-authentication-with-rhsso.adoc:
```diff
@@ -0,0 +1,172 @@
+Declarative resolvers to override the default resolver: `emailLocalPartMatchingUserEntityName`.
+The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed.
+
+.`app-config-rhdh.yaml` fragment with optional `callbackURL` field
```
```suggestion
.Example `app-config-rhdh.yaml` fragment with optional `callbackURL` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+[id="provisioning-users-from-rhsso-to-the-software-catalog"]
+= Provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog
```
Consider adding an introduction to this section.

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+
+.Procedure
+
+* To enable RHSSO member discovery, edit your custom {product-short} config map, such as app-config-rhdh, and add following lines to the app-config-rhdh.yaml content:
```
```suggestion
* To enable RHSSO member discovery, edit your custom {product-short} ConfigMap, such as `app-config-rhdh`, and add the following lines to the `app-config-rhdh.yaml` content:
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
++
+--
+[id=keycloakOrgProviderId]
+.`app-config.yaml` fragment with mandatory `keycloakOrg` fields
```
```suggestion
.Example `app-config.yaml` fragment with mandatory `keycloakOrg` fields
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+.`app-config.yaml` fragment with mandatory `keycloakOrg` fields
+[source,yaml]
+----
+dangerouslyAllowSignInWithoutUserInCatalog: false
```
```suggestion
dangerouslyAllowSignInWithoutUserInCatalog: false <1>
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
+----
+
+`dangerouslyAllowSignInWithoutUserInCatalog: false`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+----
+
+`dangerouslyAllowSignInWithoutUserInCatalog: false`::
+ Allow authentication only for users present in the {product-short} software catalog.
```
```suggestion
 <1> Allow authentication only for users in the {product-short} software catalog.
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+  providers:
+    keycloakOrg:
+      default:
+        baseUrl: ${AUTH_OIDC_METADATA_URL}
```
```suggestion
        baseUrl: ${AUTH_OIDC_METADATA_URL} <2>
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+`dangerouslyAllowSignInWithoutUserInCatalog: false`::
+ Allow authentication only for users present in the {product-short} software catalog.
+
+`baseUrl`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+ Allow authentication only for users present in the {product-short} software catalog.
+
+`baseUrl`::
+Your RHSSO server URL, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```
```suggestion
<2> Your RHSSO server URL, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+    keycloakOrg:
+      default:
+        baseUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
```
```suggestion
        clientId: ${AUTH_OIDC_CLIENT_ID} <3>
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+`baseUrl`::
+Your RHSSO server URL, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
+
+`clientId`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+Your RHSSO server URL, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
+
+`clientId`::
+Your {product-short} application client ID in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```
```suggestion
<3> Your {product-short} application client ID in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+`clientId`::
+Your {product-short} application client ID in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
+
+`clientSecret`::
```
```suggestion

```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+Your {product-short} application client ID in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
+
+`clientSecret`::
+Your {product-short} application client secret in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```
```suggestion
<4> Your {product-short} application client secret in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+      default:
+        baseUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
```
```suggestion
        clientSecret: ${AUTH_OIDC_CLIENT_SECRET} <4>
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+`clientSecret`::
+Your {product-short} application client secret in RHSSO, defined when xref:enabling-authentication-with-rhsso[enabling authentication with RHSSO].
+
+Optional: Consider adding following optional  fields:
```
```suggestion
Optional: Consider adding the following optional fields:
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+Realm to synchronize.
+Default value: `master`.
++
+.`app-config.yaml` fragment with optional `realm` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `realm` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+Realm used to authenticate.
+Default value: `master`.
++
+.`app-config.yaml` fragment with optional `loginRealm` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `loginRealm` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+User number to query simultaneously.
+Default value: `100`.
++
+.`app-config.yaml` fragment with optional `userQuerySize` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `userQuerySize` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+Group number to query simultaneously.
+Default value: `100`.
++
+.`app-config.yaml` fragment with optional `groupQuerySize` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `groupQuerySize` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+To specify custom schedule frequency.
+Supports cron, ISO duration, and "human duration" as used in code.
++
+.`app-config.yaml` fragment with optional `schedule.frequency` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `schedule.frequency` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+To specify custom timeout.
+Supports ISO duration and "human duration" as used in code.
++
+.`app-config.yaml` fragment with optional `schedule.timeout` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `schedule.timeout` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+To specify custom initial delay.
+Supports ISO duration and "human duration" as used in code.
++
+.`app-config.yaml` fragment with optional `schedule.initialDelay` field
```
```suggestion
.Example `app-config.yaml` fragment with optional `schedule.initialDelay` field
```

hmanwani-rh commented on modules/authentication/proc-provisioning-users-from-rhsso-to-the-software-catalog.adoc:
```diff
@@ -0,0 +1,153 @@
+
+.Verification
+
+. Check the console logs to verify the synchronization has been completed.
```
```suggestion
. Check the console logs to verify that the synchronization is completed.
```
